kernel-ewasm/cap9-std/src/syscalls.rs

```diff
@@ -395,20 +442,25 @@ impl Serialize for Call {
     }
 }
 
-impl Deserialize for Payload {
-    type Error = io::Error;
+impl Deserialize<u8> for Payload {
+    type Error = cap9_core::Error;
 
-    fn deserialize<R: io::Read>(reader: &mut R) -> Result<Self, Self::Error> {
+    fn deserialize<R: cap9_core::Read<u8>>(reader: &mut R) -> Result<Self, Self::Error> {
```

Comment:
This will be replaced I assume on rebase.

Reply:
Replaced with what? This branches off the refactor PR.

Reply:
Ah, you mean payload will be updated? Yes. This comment window didn't show much context.